src/sinks/appsignal/integration_tests.rs

```diff
@@ -0,0 +1,67 @@
+use futures::stream;
+use indoc::indoc;
+use vector_core::event::{BatchNotifier, BatchStatus, Event, Metric, MetricKind, MetricValue};
+
+use crate::{
+    config::SinkConfig,
+    sinks::appsignal::AppsignalSinkConfig,
+    sinks::util::test::load_sink,
+    test_util::{
+        components::{assert_sink_compliance, run_and_assert_sink_compliance, SINK_TAGS},
+        generate_lines_with_stream, map_event_batch_stream,
+    },
+};
+
```

Comment:
Thanks for adding integration tests!

It would be great to see an error path test case, if possible.